File: backend-user/src/main/java/com/matchpet/backend_user/dto/user/UserProfileResponse.java
package com.matchpet.backend_user.dto.user;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;
import java.util.Set;

@Data
@Builder
public class UserProfileResponse {

    private Integer usuarioId;
    private String email;

    // (Campos separados en lugar de 'nombreCompleto')
    private String nombre;
    private String apellidoPaterno;
    private String apellidoMaterno;

    private String telefono;
    private Set<String> roles;

    // (Añadimos los campos del perfil para una respuesta completa)
    private LocalDate fechaNacimiento;
    private String direccion;
    private String ciudad;
    private String pais;

    // (Puedes añadir más campos del perfil si los necesitas)
}

File: frontend-movile/app/src/main/java/com/example/matchpet/data/model/UserProfileResponse.kt
package com.example.matchpet.data.model

import com.google.gson.annotations.SerializedName

data class UserProfileResponse(
    val usuarioId: Int,
    val email: String,
    val nombre: String,
    val apellidoPaterno: String,
    val apellidoMaterno: String,
    val telefono: String?,
    val fechaNacimiento: String,
    val direccion: String?,
    val ciudad: String?,
    val pais: String?,
    val roles: List<String>,

    // ⭐ ESTE ES EL CAMPO QUE TE FALTABA
    @SerializedName("refugio")
    val refugio: RefugioProfileResponse?
)
